apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -88,57 +100,84 @@ const initialState: AichatState = {
 // output a message to the chat window with the list of customizations that were updated.
 export const updateAiCustomization = createAsyncThunk(
   'aichat/updateAiCustomization',
+  async (_, thunkAPI) => {
+    await updateAiCustomizationShared(thunkAPI);
+  }
+);
+
+export const publishModel = createAsyncThunk(
+  'aichat/publishModelCard',
+  async (_, thunkAPI) => {
+    thunkAPI.dispatch(setHasPublished(true));
+    await updateAiCustomizationShared(thunkAPI);
+    thunkAPI.dispatch(setViewMode(ViewMode.PRESENTATION));
+  }
+);
+
+export const saveModelCard = createAsyncThunk(
```

Comment:
nice refactoring! mind dropping some comments around how the code organized between these various thunks and the shared function?

Reply:
Cool refactoring - learned a lot here.